caffeine/src/main/java/com/github/benmanes/caffeine/cache/AsyncCache.java

```diff
@@ -71,7 +73,9 @@ public interface AsyncCache<K, V> {
    * @return the current (existing or computed) future value associated with the specified key
    * @throws NullPointerException if the specified key or mappingFunction is null
    */
-  CompletableFuture<V> get(K key, Function<? super K, ? extends V> mappingFunction);
+  @NullUnmarked
+  @NonNull CompletableFuture<V> get(
+      @NonNull K key, @NonNull Function<? super @NonNull K, ? extends V> mappingFunction);
```

Comment:
Actually, maybe I should include `@Nullable` for the output type of the functions? That would work around the Kotlin bug that I mentioned, and it would still be accurate. All that we really care about is keeping the return type convenient. I may give that a try shortly....

Reply:
I agree, if we need to use `@NullUnmarked` then we should be fully explicit because it is confusing to have defaults differ between methods.

Reply:
OK, I've added `@Nullable` to all the function output types. As expected, this appears to be helping with the Kotlin issues (though my testing is still underway).